UserController.js

```diff
@@ -64,47 +64,87 @@ class UserController {
 
   async knockOnFriends(req, res) {
     try {
-      const { name, age, sex } = req.body;
-      const user = await User.create({ name, age, sex, flag: 'not-approv'});
-      res.json(user);
+      const { id } = req.params;
+      const { _id } = req.body;
+      if (!id && _id2 && (id1 != id2)) {
+        res.status(400).json({ message: "Укажите id!" });
+      }
+
+      const user = await User.findById(id);
+      if (user.request_id.includes(_id)) {
+        res.status(400).json({ message: "Вы уже подали заявку" });
+      }else if (user.friends_id.includes(_id)) {
+        res.status(400).json({ message: "Вы уже в друзьях" });
+      }
+
+      user.request_id.push(_id);
+      const updatedUser = await User.findByIdAndUpdate(user._id, user, { new: true });
+      return res.json(updatedUser);
     } catch (e) {
       res.status(500).json(e);
     }
   }
 
-  async showFriends(req, res) {
+  async addUser(req, res) {
```

Comment:
this name is also absolutely awful. please read the paragraph about naming variables in the book you are reading now and then changes names